File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/ServerTypes/Prescriptions/PrescriptionsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DrugListFormatter } from "../../Prescriptions/Prescriptions/DrugListFormatter";
import { PrescriptionsRow } from "./PrescriptionsRow";

export interface PrescriptionsColumns {
    PrescriptionId: Column<PrescriptionsRow>;
    ExaminationComplaint: Column<PrescriptionsRow>;
    DoctorName: Column<PrescriptionsRow>;
    PatientName: Column<PrescriptionsRow>;
    PrescriptionNote: Column<PrescriptionsRow>;
    PrescriptionDate: Column<PrescriptionsRow>;
    DrugList: Column<PrescriptionsRow>;
}

export class PrescriptionsColumns extends ColumnsBase<PrescriptionsRow> {
    static readonly columnsKey = 'Prescriptions.Prescriptions';
    static readonly Fields = fieldsProxy<PrescriptionsColumns>();
}

[DrugListFormatter]; // referenced types

File: MuayeneYonetimPortali/MuayeneYonetimPortali.Web/Modules/Prescriptions/Prescriptions/PrescriptionsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace MuayeneYonetimPortali.Prescriptions.Columns;

[ColumnsScript("Prescriptions.Prescriptions")]
[BasedOnRow(typeof(PrescriptionsRow), CheckNames = true)]
public class PrescriptionsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId")]
    public int PrescriptionId { get; set; }
    [EditLink]
    public string ExaminationComplaint { get; set; }
    [EditLink]
    public string DoctorName { get; set; }
    [EditLink]
    public string PatientName { get; set; }
    [EditLink]
    public string PrescriptionNote { get; set; }
    [EditLink]
    public DateTime PrescriptionDate { get; set; }
    [Width(250), DrugListFormatter, EditLink]
    public List<int> DrugList { get; set; }
}